src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -44,6 +55,45 @@ public function iOpenProductPage(ProductInterface $product): void
         $this->client->show($product->getCode());
     }
 
+    /**
+     * @When /^I browse products from (taxon "([^"]+)")$/
+     */
+    public function iBrowseProductsFromTaxon(TaxonInterface $taxon): void
+    {
+        $this->client->index();
+        $this->client->addFilter('productTaxons.taxon.code', $taxon->getCode());
+        $this->client->filter();
+    }
+
+    /**
+     * @Then /^I should see the product price ("[^"]+")$/
+     */
+    public function iShouldSeeTheProductPrice(int $price): void
+    {
+        Assert::true(
+            $this->hasProductWithPriceInChannel(
+                [$this->responseChecker->getResponseContent($this->client->getLastResponse())],
+                $price,
+                $this->sharedStorage->get('channel'),
+            )
+        );
+    }
+
+    /**
+     * @Then /^I should see the (product "[^"]+") with price ("[^"]+")$/
+     */
+    public function iShouldSeeTheProductWithPrice(ProductInterface $product, int $price): void
+    {
+        Assert::true(
+            $this->hasProductWithPriceInChannel(
+                $this->responseChecker->getCollection($this->client->getLastResponse()),
+                $price,
+                $this->sharedStorage->get('channel'),
+                $product->getCode()
+            )
```

Comment:
```suggestion
            ),
            sprintf("there is no product with %s code and %s price", $product->getCode(), $price)
```
when this assertion fails, a custom message will be helpful to understand what went wrong.